modules/social_features/social_group/modules/social_group_request/social_group_request.module

```diff
@@ -302,26 +302,28 @@ function social_group_request_preprocess_activity(array &$variables) {
     /** @var Drupal\message\Entity\Message $message */
     $message = $activity->field_activity_message->entity;
 
-    if (!$message->get('template')->isEmpty()) {
-      $message_template = $message->template->target_id;
-
-      if ($message_template === 'request_to_join_a_group') {
-        $activity_entity = $activity->getRelatedEntity();
-
-        if ($activity_entity instanceof GroupContentInterface) {
-          $variables['full_url'] = Url::fromRoute('view.group_pending_members.membership_requests', [
-            'arg_0' => $activity_entity->getGroup()->id(),
-          ]);
+    if (NULL !== $message) {
```

Comment:
Hi, @DmytroRovovyi Thanks for your improvements!

Let's check if `$message` is an instance of **Message** since we expect exactly that entity:

```
/**
 * Implements hook_preprocess_activity().
 */
function social_group_request_preprocess_activity(array &$variables) {
  /** @var \Drupal\activity_creator\Entity\Activity $entity */
  $activity = $variables['elements']['#activity'];

  if (!$activity->get('field_activity_message')->isEmpty()) {
    $message_storage = \Drupal::entityTypeManager()->getStorage('message');
    /** @var Drupal\message\Entity\Message $message */
    $message = $message_storage->load($activity->field_activity_message->target_id);

    if ($message instanceof Message) {
     ...
```

Reply:
@Ressinel yes I think it's a good idea

Reply:
@DmytroRovovyi By any chance you can look at this again?

Reply:
@navneet0693 I'll try to look into this issue as soon as possible.

Reply:
@Ressinel @navneet0693  changed check for $message made check for whether $message is an instance